libs/wallet/src/web3-react/updater.ts

```diff
@@ -86,19 +90,39 @@ function _useSafeInfo(walletInfo: WalletInfo): GnosisSafeInfo | undefined {
       }
     }
 
-    const interval = setInterval(update, SAFE_INFO_UPDATE_INTERVAL)
+    const updateSafeAppsSdkInfo = async () => {
+      if (safeAppsSdk) {
+        const appsSdkSafeInfo = await safeAppsSdk.safe.getInfo()
+        setSafeInfo((prevSafeInfo) => {
+          // only update the isReadyOnly flag if we have safe info returned from the api kit
+          if (!prevSafeInfo) return
+          return {
+            ...prevSafeInfo,
+            isReadOnly: appsSdkSafeInfo.isReadOnly,
+          }
+        })
+      }
+    }
 
-    update()
+    const safeInfoInterval = setInterval(updateSafeInfo, SAFE_INFO_UPDATE_INTERVAL)
+    const safeAppsSdkInfoInterval = setInterval(updateSafeAppsSdkInfo, SAFE_APPS_SDK_INFO_UPDATE_INTERVAL)
 
-    return () => clearInterval(interval)
-  }, [setSafeInfo, chainId, account, provider, isReadOnly])
+    updateSafeInfo()
+    updateSafeAppsSdkInfo()
```

Comment:
Is it possible for them to be out of sync in regards to `isReadOnly`?
For example, what would happen if the sdkInfo returns `false`, then another update from `getSafeInfo` runs and it's marked as `true`?

Reply:
Yes, I thought about this. Since both updates use the functional update pattern, then it is guaranteed that at the time of running those functions would have the most up to date previous state. 

This doesn't really help with the potential race condition, but I think that it should be fine:
- updateSafeInfo doesn't touch the isReadOnly flag. it sets the previous state, then overrides it with the values from getSafeInfo, note that getSafeInfo doesn't return an isReadOnly flag, so in reality the only function that can update this state is updateIsReadOnlyFromSafeAppsSdkInfo. Makes sense?

Reply:
Ah ok. Then it's fine.